conn.go

```diff
@@ -478,20 +508,68 @@ func (c *Conn) handleRcpt(arg string) {
 		return
 	}
 
-	if (len(arg) < 4) || (strings.ToUpper(arg[0:3]) != "TO:") {
-		c.WriteResponse(501, EnhancedCode{5, 5, 2}, "Was expecting RCPT arg syntax of TO:<address>")
+	if len(arg) < 4 || strings.ToUpper(arg[0:3]) != "TO:" {
+		c.WriteResponse(501, EnhancedCode{5, 5, 1}, "Was expecting TO arg syntax of TO:<address>")
 		return
 	}
-
-	// TODO: This trim is probably too forgiving
-	recipient := strings.Trim(arg[3:], "<> ")
+	toArgs := strings.Split(strings.Trim(arg[3:], " "), " ")
+	if c.server.Strict {
+		if !strings.HasPrefix(toArgs[0], "<") || !strings.HasSuffix(toArgs[0], ">") {
+			c.WriteResponse(501, EnhancedCode{5, 5, 1}, "Was expecting TO arg syntax of TO:<address>")
+			return
+		}
+	}
+	recipient := toArgs[0]
+	if recipient == "" {
+		c.WriteResponse(501, EnhancedCode{5, 5, 2}, "Was expecting MAIL arg syntax of FROM:<address>")
+		return
+	}
+	recipient = strings.Trim(recipient, "<>")
```

Comment:
Can we keep the TODO about this trim being not quite what we want?

Reply:
I also don't like the existing TODO, and proposed to use `mail.ParseAddress`. How is it implemented in postfix/exim mail servers? Shall we introduce a similar function?